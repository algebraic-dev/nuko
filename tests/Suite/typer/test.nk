type A b c =
  | F b
  | D b c
  | E c