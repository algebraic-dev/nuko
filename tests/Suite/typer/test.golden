✓ That
 └Prod
   ├List
   | └TypeDecl
   |   ├tyName: (ReId A 0:5-0:6)
   |   ├tyArgs:
   |   | └List (ReId c 0:9-0:10) (ReId b 0:7-0:8)
   |   └tyDecl:
   |     └TypeSum
   |       └NonEmpty
   |         ├Prod
   |         | ├ReId F 1:4-1:5
   |         | └List (Type [No 0])
   |         └Prod
   |           ├ReId D 2:4-2:5
   |           └List (Type [No 1]) (Type [No 0])
   └TypeSpace
     ├_tsTypes:
     | └HashMap
     |   └Prod
     |     ├A
     |     └Kind (^b.0 -> (^c.0 -> *))
     ├_tsFields: HashMap
     └_tsConstructors:
       └HashMap
         ├Prod
         | ├F
         | └Type ([No 0] -> ((A b) c))
         └Prod
           ├D
           └Type ([No 0] -> ([No 1] -> ((A b) c)))
